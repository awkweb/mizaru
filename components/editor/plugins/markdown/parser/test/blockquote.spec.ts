import Parser from '../'

// @ts-ignore
const { mkHeadings } = prosemirror
const headings = mkHeadings()

for (const syntax of ['>', '> ']) {
    describe(syntax, () => {
        test('empty', () => {
            const doc = `${syntax}`
            const out = Parser.parse(doc)
            expect(out).toMatchSnapshot()
        })

        test('single word', () => {
            const doc = `${syntax}foo`
            const out = Parser.parse(doc)
            expect(out).toMatchSnapshot()
        })

        test('multiple words', () => {
            const doc = `${syntax}foo bar baz`
            const out = Parser.parse(doc)
            expect(out).toMatchSnapshot()
        })

        test('backslash', () => {
            const doc = `${syntax}foo\\ bar baz`
            const out = Parser.parse(doc)
            expect(out).toMatchSnapshot()
        })

        test('split across multiple lines', () => {
            const doc = `${syntax}foo\nbar\nbaz`
            const out = Parser.parse(doc)
            expect(out).toMatchSnapshot()
        })

        describe('with nested', () => {
            test('delete', () => {
                const doc = `${syntax}foo ~~bar~~ baz`
                const out = Parser.parse(doc)
                expect(out).toMatchSnapshot()
            })

            test('emphasis', () => {
                const doc = `${syntax}foo *bar* baz`
                const out = Parser.parse(doc)
                expect(out).toMatchSnapshot()
            })

            test('inlineCode', () => {
                const doc = `${syntax}foo \`bar\` baz`
                const out = Parser.parse(doc)
                expect(out).toMatchSnapshot()
            })

            test('strong', () => {
                const doc = `${syntax}foo **bar** baz`
                const out = Parser.parse(doc)
                expect(out).toMatchSnapshot()
            })

            for (const heading of headings) {
                test(heading.tag, () => {
                    const doc = `${syntax}${heading.syntax} foo bar baz`
                    const out = Parser.parse(doc)
                    expect(out).toMatchSnapshot()
                })
            }
        })

        describe('with whitespace', () => {
            test('leading', () => {
                const doc = `   ${syntax}foo bar baz`
                const out = Parser.parse(doc)
                expect(out).toMatchSnapshot()
            })

            test('inner', () => {
                const doc = `${syntax}   foo bar baz`
                const out = Parser.parse(doc)
                expect(out).toMatchSnapshot()
            })

            test('leading and inner', () => {
                const doc = `   ${syntax}   foo bar baz`
                const out = Parser.parse(doc)
                expect(out).toMatchSnapshot()
            })
        })
    })
}
