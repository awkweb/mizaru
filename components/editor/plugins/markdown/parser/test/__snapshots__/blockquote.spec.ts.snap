// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>  backslash 1`] = `
Object {
  "counter": 18,
  "decorations": Array [
    Object {
      "from": 2,
      "to": 4,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "from": 1,
      "marks": Array [],
      "to": 17,
      "type": "paragraph",
    },
    Object {
      "from": 0,
      "to": 18,
      "type": "blockquote",
    },
  ],
}
`;

exports[`>  empty 1`] = `
Object {
  "counter": 2,
  "decorations": Array [
    Object {
      "from": 2,
      "to": 4,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "from": 0,
      "to": 2,
      "type": "blockquote",
    },
  ],
}
`;

exports[`>  multiple words 1`] = `
Object {
  "counter": 17,
  "decorations": Array [
    Object {
      "from": 2,
      "to": 4,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "from": 1,
      "marks": Array [],
      "to": 16,
      "type": "paragraph",
    },
    Object {
      "from": 0,
      "to": 17,
      "type": "blockquote",
    },
  ],
}
`;

exports[`>  single word 1`] = `
Object {
  "counter": 9,
  "decorations": Array [
    Object {
      "from": 2,
      "to": 4,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "from": 1,
      "marks": Array [],
      "to": 8,
      "type": "paragraph",
    },
    Object {
      "from": 0,
      "to": 9,
      "type": "blockquote",
    },
  ],
}
`;

exports[`>  split across multiple lines 1`] = `
Object {
  "counter": 19,
  "decorations": Array [
    Object {
      "from": 2,
      "to": 4,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "from": 1,
      "marks": Array [],
      "to": 18,
      "type": "paragraph",
    },
    Object {
      "from": 0,
      "to": 19,
      "type": "blockquote",
    },
  ],
}
`;

exports[`>  with nested delete 1`] = `
Object {
  "counter": 21,
  "decorations": Array [
    Object {
      "from": 8,
      "to": 10,
      "type": "syntax",
    },
    Object {
      "from": 13,
      "to": 15,
      "type": "syntax",
    },
    Object {
      "from": 2,
      "to": 4,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "from": 1,
      "marks": Array [
        Object {
          "from": 8,
          "to": 15,
          "type": "delete",
        },
      ],
      "to": 20,
      "type": "paragraph",
    },
    Object {
      "from": 0,
      "to": 21,
      "type": "blockquote",
    },
  ],
}
`;

exports[`>  with nested emphasis 1`] = `
Object {
  "counter": 19,
  "decorations": Array [
    Object {
      "from": 8,
      "to": 9,
      "type": "syntax",
    },
    Object {
      "from": 12,
      "to": 13,
      "type": "syntax",
    },
    Object {
      "from": 2,
      "to": 4,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "from": 1,
      "marks": Array [
        Object {
          "from": 8,
          "to": 13,
          "type": "emphasis",
        },
      ],
      "to": 18,
      "type": "paragraph",
    },
    Object {
      "from": 0,
      "to": 19,
      "type": "blockquote",
    },
  ],
}
`;

exports[`>  with nested h1 1`] = `
Object {
  "counter": 19,
  "decorations": Array [
    Object {
      "from": 4,
      "to": 6,
      "type": "syntax",
    },
    Object {
      "from": 2,
      "to": 4,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "attrs": Object {
        "level": 1,
      },
      "from": 1,
      "marks": Array [],
      "to": 18,
      "type": "heading",
    },
    Object {
      "from": 0,
      "to": 19,
      "type": "blockquote",
    },
  ],
}
`;

exports[`>  with nested h2 1`] = `
Object {
  "counter": 20,
  "decorations": Array [
    Object {
      "from": 4,
      "to": 7,
      "type": "syntax",
    },
    Object {
      "from": 2,
      "to": 4,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "attrs": Object {
        "level": 2,
      },
      "from": 1,
      "marks": Array [],
      "to": 19,
      "type": "heading",
    },
    Object {
      "from": 0,
      "to": 20,
      "type": "blockquote",
    },
  ],
}
`;

exports[`>  with nested h3 1`] = `
Object {
  "counter": 21,
  "decorations": Array [
    Object {
      "from": 4,
      "to": 8,
      "type": "syntax",
    },
    Object {
      "from": 2,
      "to": 4,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "attrs": Object {
        "level": 3,
      },
      "from": 1,
      "marks": Array [],
      "to": 20,
      "type": "heading",
    },
    Object {
      "from": 0,
      "to": 21,
      "type": "blockquote",
    },
  ],
}
`;

exports[`>  with nested h4 1`] = `
Object {
  "counter": 22,
  "decorations": Array [
    Object {
      "from": 4,
      "to": 9,
      "type": "syntax",
    },
    Object {
      "from": 2,
      "to": 4,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "attrs": Object {
        "level": 4,
      },
      "from": 1,
      "marks": Array [],
      "to": 21,
      "type": "heading",
    },
    Object {
      "from": 0,
      "to": 22,
      "type": "blockquote",
    },
  ],
}
`;

exports[`>  with nested h5 1`] = `
Object {
  "counter": 23,
  "decorations": Array [
    Object {
      "from": 4,
      "to": 10,
      "type": "syntax",
    },
    Object {
      "from": 2,
      "to": 4,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "attrs": Object {
        "level": 5,
      },
      "from": 1,
      "marks": Array [],
      "to": 22,
      "type": "heading",
    },
    Object {
      "from": 0,
      "to": 23,
      "type": "blockquote",
    },
  ],
}
`;

exports[`>  with nested h6 1`] = `
Object {
  "counter": 24,
  "decorations": Array [
    Object {
      "from": 4,
      "to": 11,
      "type": "syntax",
    },
    Object {
      "from": 2,
      "to": 4,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "attrs": Object {
        "level": 6,
      },
      "from": 1,
      "marks": Array [],
      "to": 23,
      "type": "heading",
    },
    Object {
      "from": 0,
      "to": 24,
      "type": "blockquote",
    },
  ],
}
`;

exports[`>  with nested inlineCode 1`] = `
Object {
  "counter": 19,
  "decorations": Array [
    Object {
      "from": 8,
      "to": 9,
      "type": "syntax",
    },
    Object {
      "from": 12,
      "to": 13,
      "type": "syntax",
    },
    Object {
      "from": 2,
      "to": 4,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "from": 1,
      "marks": Array [
        Object {
          "from": 8,
          "to": 13,
          "type": "inlineCode",
        },
      ],
      "to": 18,
      "type": "paragraph",
    },
    Object {
      "from": 0,
      "to": 19,
      "type": "blockquote",
    },
  ],
}
`;

exports[`>  with nested strong 1`] = `
Object {
  "counter": 21,
  "decorations": Array [
    Object {
      "from": 8,
      "to": 10,
      "type": "syntax",
    },
    Object {
      "from": 13,
      "to": 15,
      "type": "syntax",
    },
    Object {
      "from": 2,
      "to": 4,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "from": 1,
      "marks": Array [
        Object {
          "from": 8,
          "to": 15,
          "type": "strong",
        },
      ],
      "to": 20,
      "type": "paragraph",
    },
    Object {
      "from": 0,
      "to": 21,
      "type": "blockquote",
    },
  ],
}
`;

exports[`>  with whitespace inner 1`] = `
Object {
  "counter": 20,
  "decorations": Array [
    Object {
      "from": 2,
      "to": 4,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "from": 1,
      "marks": Array [],
      "to": 19,
      "type": "paragraph",
    },
    Object {
      "from": 0,
      "to": 20,
      "type": "blockquote",
    },
  ],
}
`;

exports[`>  with whitespace leading 1`] = `
Object {
  "counter": 20,
  "decorations": Array [
    Object {
      "from": 2,
      "to": 7,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "from": 1,
      "marks": Array [],
      "to": 19,
      "type": "paragraph",
    },
    Object {
      "from": 0,
      "to": 20,
      "type": "blockquote",
    },
  ],
}
`;

exports[`>  with whitespace leading and inner 1`] = `
Object {
  "counter": 23,
  "decorations": Array [
    Object {
      "from": 2,
      "to": 7,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "from": 1,
      "marks": Array [],
      "to": 22,
      "type": "paragraph",
    },
    Object {
      "from": 0,
      "to": 23,
      "type": "blockquote",
    },
  ],
}
`;

exports[`> backslash 1`] = `
Object {
  "counter": 17,
  "decorations": Array [
    Object {
      "from": 2,
      "to": 3,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "from": 1,
      "marks": Array [],
      "to": 16,
      "type": "paragraph",
    },
    Object {
      "from": 0,
      "to": 17,
      "type": "blockquote",
    },
  ],
}
`;

exports[`> empty 1`] = `
Object {
  "counter": 3,
  "decorations": Array [],
  "nodes": Array [
    Object {
      "from": 0,
      "marks": Array [],
      "to": 3,
      "type": "paragraph",
    },
  ],
}
`;

exports[`> multiple words 1`] = `
Object {
  "counter": 16,
  "decorations": Array [
    Object {
      "from": 2,
      "to": 3,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "from": 1,
      "marks": Array [],
      "to": 15,
      "type": "paragraph",
    },
    Object {
      "from": 0,
      "to": 16,
      "type": "blockquote",
    },
  ],
}
`;

exports[`> single word 1`] = `
Object {
  "counter": 8,
  "decorations": Array [
    Object {
      "from": 2,
      "to": 3,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "from": 1,
      "marks": Array [],
      "to": 7,
      "type": "paragraph",
    },
    Object {
      "from": 0,
      "to": 8,
      "type": "blockquote",
    },
  ],
}
`;

exports[`> split across multiple lines 1`] = `
Object {
  "counter": 18,
  "decorations": Array [
    Object {
      "from": 2,
      "to": 3,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "from": 1,
      "marks": Array [],
      "to": 17,
      "type": "paragraph",
    },
    Object {
      "from": 0,
      "to": 18,
      "type": "blockquote",
    },
  ],
}
`;

exports[`> with nested delete 1`] = `
Object {
  "counter": 20,
  "decorations": Array [
    Object {
      "from": 7,
      "to": 9,
      "type": "syntax",
    },
    Object {
      "from": 12,
      "to": 14,
      "type": "syntax",
    },
    Object {
      "from": 2,
      "to": 3,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "from": 1,
      "marks": Array [
        Object {
          "from": 7,
          "to": 14,
          "type": "delete",
        },
      ],
      "to": 19,
      "type": "paragraph",
    },
    Object {
      "from": 0,
      "to": 20,
      "type": "blockquote",
    },
  ],
}
`;

exports[`> with nested emphasis 1`] = `
Object {
  "counter": 18,
  "decorations": Array [
    Object {
      "from": 7,
      "to": 8,
      "type": "syntax",
    },
    Object {
      "from": 11,
      "to": 12,
      "type": "syntax",
    },
    Object {
      "from": 2,
      "to": 3,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "from": 1,
      "marks": Array [
        Object {
          "from": 7,
          "to": 12,
          "type": "emphasis",
        },
      ],
      "to": 17,
      "type": "paragraph",
    },
    Object {
      "from": 0,
      "to": 18,
      "type": "blockquote",
    },
  ],
}
`;

exports[`> with nested h1 1`] = `
Object {
  "counter": 18,
  "decorations": Array [
    Object {
      "from": 3,
      "to": 5,
      "type": "syntax",
    },
    Object {
      "from": 2,
      "to": 3,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "attrs": Object {
        "level": 1,
      },
      "from": 1,
      "marks": Array [],
      "to": 17,
      "type": "heading",
    },
    Object {
      "from": 0,
      "to": 18,
      "type": "blockquote",
    },
  ],
}
`;

exports[`> with nested h2 1`] = `
Object {
  "counter": 19,
  "decorations": Array [
    Object {
      "from": 3,
      "to": 6,
      "type": "syntax",
    },
    Object {
      "from": 2,
      "to": 3,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "attrs": Object {
        "level": 2,
      },
      "from": 1,
      "marks": Array [],
      "to": 18,
      "type": "heading",
    },
    Object {
      "from": 0,
      "to": 19,
      "type": "blockquote",
    },
  ],
}
`;

exports[`> with nested h3 1`] = `
Object {
  "counter": 20,
  "decorations": Array [
    Object {
      "from": 3,
      "to": 7,
      "type": "syntax",
    },
    Object {
      "from": 2,
      "to": 3,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "attrs": Object {
        "level": 3,
      },
      "from": 1,
      "marks": Array [],
      "to": 19,
      "type": "heading",
    },
    Object {
      "from": 0,
      "to": 20,
      "type": "blockquote",
    },
  ],
}
`;

exports[`> with nested h4 1`] = `
Object {
  "counter": 21,
  "decorations": Array [
    Object {
      "from": 3,
      "to": 8,
      "type": "syntax",
    },
    Object {
      "from": 2,
      "to": 3,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "attrs": Object {
        "level": 4,
      },
      "from": 1,
      "marks": Array [],
      "to": 20,
      "type": "heading",
    },
    Object {
      "from": 0,
      "to": 21,
      "type": "blockquote",
    },
  ],
}
`;

exports[`> with nested h5 1`] = `
Object {
  "counter": 22,
  "decorations": Array [
    Object {
      "from": 3,
      "to": 9,
      "type": "syntax",
    },
    Object {
      "from": 2,
      "to": 3,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "attrs": Object {
        "level": 5,
      },
      "from": 1,
      "marks": Array [],
      "to": 21,
      "type": "heading",
    },
    Object {
      "from": 0,
      "to": 22,
      "type": "blockquote",
    },
  ],
}
`;

exports[`> with nested h6 1`] = `
Object {
  "counter": 23,
  "decorations": Array [
    Object {
      "from": 3,
      "to": 10,
      "type": "syntax",
    },
    Object {
      "from": 2,
      "to": 3,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "attrs": Object {
        "level": 6,
      },
      "from": 1,
      "marks": Array [],
      "to": 22,
      "type": "heading",
    },
    Object {
      "from": 0,
      "to": 23,
      "type": "blockquote",
    },
  ],
}
`;

exports[`> with nested inlineCode 1`] = `
Object {
  "counter": 18,
  "decorations": Array [
    Object {
      "from": 7,
      "to": 8,
      "type": "syntax",
    },
    Object {
      "from": 11,
      "to": 12,
      "type": "syntax",
    },
    Object {
      "from": 2,
      "to": 3,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "from": 1,
      "marks": Array [
        Object {
          "from": 7,
          "to": 12,
          "type": "inlineCode",
        },
      ],
      "to": 17,
      "type": "paragraph",
    },
    Object {
      "from": 0,
      "to": 18,
      "type": "blockquote",
    },
  ],
}
`;

exports[`> with nested strong 1`] = `
Object {
  "counter": 20,
  "decorations": Array [
    Object {
      "from": 7,
      "to": 9,
      "type": "syntax",
    },
    Object {
      "from": 12,
      "to": 14,
      "type": "syntax",
    },
    Object {
      "from": 2,
      "to": 3,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "from": 1,
      "marks": Array [
        Object {
          "from": 7,
          "to": 14,
          "type": "strong",
        },
      ],
      "to": 19,
      "type": "paragraph",
    },
    Object {
      "from": 0,
      "to": 20,
      "type": "blockquote",
    },
  ],
}
`;

exports[`> with whitespace inner 1`] = `
Object {
  "counter": 19,
  "decorations": Array [
    Object {
      "from": 2,
      "to": 4,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "from": 1,
      "marks": Array [],
      "to": 18,
      "type": "paragraph",
    },
    Object {
      "from": 0,
      "to": 19,
      "type": "blockquote",
    },
  ],
}
`;

exports[`> with whitespace leading 1`] = `
Object {
  "counter": 19,
  "decorations": Array [
    Object {
      "from": 2,
      "to": 6,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "from": 1,
      "marks": Array [],
      "to": 18,
      "type": "paragraph",
    },
    Object {
      "from": 0,
      "to": 19,
      "type": "blockquote",
    },
  ],
}
`;

exports[`> with whitespace leading and inner 1`] = `
Object {
  "counter": 22,
  "decorations": Array [
    Object {
      "from": 2,
      "to": 7,
      "type": "syntax",
    },
  ],
  "nodes": Array [
    Object {
      "from": 1,
      "marks": Array [],
      "to": 21,
      "type": "paragraph",
    },
    Object {
      "from": 0,
      "to": 22,
      "type": "blockquote",
    },
  ],
}
`;
